.github/workflows/generate_ebooks.yml

```
@@ -0,0 +1,77 @@
+######################################
+## Custom Web Almanac GitHub action ##
+######################################
+#
+# Generate the ebooks when this GitHub Action is run manually
+# This should be run everytime before release
+#
+name: Generate Ebooks
+
+env:
+  # Update periodically from https://www.princexml.com/latest/
+  PRINCE_PACKAGE: 'prince_20200728-1_ubuntu20.04_amd64.deb'
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build:
+    # Prince needs ubuntu 20 and ubuntu-latest is only on 18
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout branch
+      uses: actions/checkout@v2
+    - name: Setup Node.js for use with actions
+      uses: actions/setup-node@v1.1.0
+      with:
+        version:  12.x
+    # Install Python 3.7 as used by Google Cloud Platform
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.7'
+    - name: Install Python requirements
+      run: |
+        echo "Installing Python requirements"
+        python -m pip install --upgrade pip
+        cd src
+        pip install -r requirements.txt
+    - name: Install node modules
+      run: |
+        echo "Running npm install"
+        cd src
+        npm install
+    - name: Install PrinceXML
+      run: |
+        echo "Installing PrinceXML"
+        cd /tmp
+        sudo apt-get install -y aptitude
+        sudo aptitude install -y gdebi
+        wget https://www.princexml.com/download/${{ env.PRINCE_PACKAGE }}
+        sudo gdebi --non-interactive ${{ env.PRINCE_PACKAGE }}
+    - name: Run website
+      run: |
+        echo "Running website"
+        cd src
+        python main.py background &
+    - name: Generating Ebooks
+      run: |
+        echo "Generating ebooks"
+        cd src
+        npm run ebooks
+    - name: Create Pull Request
+      id: cpr
+      uses: peter-evans/create-pull-request@v3
+      with:
+        title: Regenerate Ebooks
+        branch-suffix: timestamp
+        commit-message: Generate Ebooks
+        body: |
+          Regenerated Ebooks through GitHub action
+          - Auto-generated by [create-pull-request][1] GitHub Action
+
+          [1]: https://github.com/peter-evans/create-pull-request
+        labels: generate chapters
+    - name: Check outputs
+      run: |
+        echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
```

This can go in a single line.

```suggestion
      run: echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
```